src/engine/strat_engine/backstore/backstore/v2.rs

```diff
@@ -0,0 +1,1626 @@
+            match pool_save.backstore.cache_tier {
+                Some(ref cache_tier_save) => {
+                    let cache_tier = match CacheTier::setup(block_mgr, cache_tier_save) {
+                        Ok(ct) => ct,
+                        Err((e, mut bdas)) => {
+                            bdas.extend(data_tier.block_mgr.into_bdas());
+                            return Err((e, bdas));
+                        }
+                    };
+
+                    let cache_device = match make_cache(pool_uuid, &cache_tier, origin, None, false)
+                    {
+                        Ok(cd) => cd,
+                        Err(e) => {
+                            return Err((
+                                e,
+                                data_tier
+                                    .block_mgr
+                                    .into_bdas()
+                                    .into_iter()
+                                    .chain(cache_tier.block_mgr.into_bdas())
+                                    .collect::<HashMap<_, _>>(),
+                            ));
+                        }
+                    };
+                    (None, Some(cache_device), Some(cache_tier), None)
+                }
+                None => {
+                    let err_msg = "Cachedevs exist, but cache metadata does not exist";
+                    return Err((
+                        StratisError::Msg(err_msg.into()),
+                        data_tier
+                            .block_mgr
+                            .into_bdas()
+                            .into_iter()
+                            .chain(block_mgr.into_bdas())
+                            .collect::<HashMap<_, _>>(),
+                    ));
+                }
+            }
+        } else {
+            let placeholder = match make_placeholder_dev(pool_uuid, &origin) {
+                Ok(pl) => pl,
+                Err(e) => return Err((e, data_tier.block_mgr.into_bdas())),
+            };
+            (Some(placeholder), None, None, Some(origin))
+        };
+
+        let metadata_enc_enabled = pool_save.features.contains(PoolFeatures::Encryption);
+        let crypt_physical_path = &once(DEVICEMAPPER_PATH)
+            .chain(once(
+                format_backstore_ids(pool_uuid, CacheRole::Cache)
+                    .0
+                    .to_string()
+                    .as_str(),
+            ))
+            .collect::<PathBuf>();
+        let enc = match (metadata_enc_enabled, unlock_method, passphrase.as_ref()) {
+            (false, Some(_), _) => return Err((
+                StratisError::Msg(
+                    "Metadata reported that encryption was not enabled but unlock method was provided"
+                        .to_string()
+                ),
+                data_tier.block_mgr.into_bdas(),
+            )),
+            (false, _, Some(_)) => return Err((
```

Comment:
I believe that this condition is already covered in ```LiminalDevices::start_pool_new()```.  I think that it would  make sense to move the ```(false, Some(_), _)``` condition into that method also to avoid setting up the cache tier unnecessarily. I think probably the ```(true, None, _)``` condition as well.